daemon/container_unix.go

```diff
@@ -914,6 +914,12 @@ func (container *Container) allocateNetwork() error {
 		if mode.IsDefault() {
 			networkName = controller.Config().Daemon.DefaultNetwork
 		}
+		if mode.IsUserDefined() {
+			if n, err := container.daemon.FindNetwork(networkName); err == nil {
+				networkName = n.Name()
```

Comment:
Please take a look at the loop at original line 923. It will cause the code to first connect to the network by id, then again by name...